### PR TITLE
Fetchable mixin

### PR DESCRIPTION
**Pull request by zinabat**
## Context
[US121368](https://rally1.rallydev.com/#/detail/userstory/441183090652?fdp=true): Creation of Fetchable mixin
[Architecture Design](https://desire2learn.atlassian.net/wiki/spaces/POL/pages/1622278478/US120980+Defining+the+architecture+of+hmc-foundation)
Adds the `Fetchable` mixin which functions as an abstract class for use with the in progress `fetch` method.

- `fetchStatus` is now an object attached to the `Fetchable` with `start()`, `cancel()`, and `done()` methods
- `observable/SirenAction` and `HypermediaState` both implement `Fetchable`
- `href(input)` method converted to `setQueryParams` method
- `header()` method converted to `headers` getter
- Snuck a `.editorconfig` file in there because it was causing problems not having one

## Quality
- added new unit tests for `FetchStatus`
- added basic tests for `Fetchable.setQueryParams()`
- manually tested with BSI branch

**Note**
There is an existing error in `master` related to `sirenAction`. I didn't want to fix it in this PR because of the current work in flight.
Screen Shot 2020-10-26 at 2 58 46 PM

### CODE REVIEW

File: state/observable/SirenAction.js [hunk 1]
```diff
@@ -1,13 +1,12 @@
+import { Fetchable } from '../Fetchable.js';
 import { Observable } from './Observable.js';
 import { performAction } from '../store.js';
-import { refreshToken } from '../token.js';
 
-export class SirenAction extends Observable {
+export class SirenAction extends Fetchable(Observable) {
 	constructor({ id: name, token, state }) {
-		super();
+		super(state?.href, token);
```
Comment (zinabat) [inline state/observable/SirenAction.js:7]: I think the state should always exist? I wasn't sure.

File: state/observable/SirenAction.js [hunk 2]
```diff
@@ -25,51 +24,43 @@ export class SirenAction extends Observable {
...
-
-		return headers;
+	get method() {
+		return this._rawSirenAction && this._rawSirenAction.method;
 	}
 
-	href(input) {
-		let url = new URL(this._rawSirenAction.href, window.location.origin);
-		if (this._rawSirenAction.method === 'GET' || this._rawSirenAction.method === 'HEAD') {
+	setBodyFromInput(input) {
```
Comment (zinabat) [inline state/observable/SirenAction.js:47]: I'd like to add tests for this, but I'm unsure what the input is supposed to look like.

File: state/observable/SirenAction.js (hunk 1 above)
Comment (m6jones) [reply state/observable/SirenAction.js:7]: defineProperties should make it always exist.

Comment (m6jones) [reply state/observable/SirenAction.js:7]: This shouldn't be the `href` that we should put to super.

Comment (zinabat) [reply state/observable/SirenAction.js:7]: Should the `href` be null then until the `setSirenEntity` gets us the `_rawSirenAction`?

Comment (tenbradley) [reply state/observable/SirenAction.js:7]: That logically makes sense to me.

File: .editorconfig [hunk 3]
```diff
@@ -0,0 +1,13 @@
+root = true
```
Comment (m6jones) [inline .editorconfig:1]: Thank you!

File: state/observable/SirenAction.js (hunk 1 above)
Comment (m6jones) [reply state/observable/SirenAction.js:7]: yup

File: state/observable/SirenAction.js (hunk 2 above)
Comment (m6jones) [reply state/observable/SirenAction.js:47]: Lets figure out tests when I setup fetch method.

**Comment by tenbradley** [comment]
Should we update the variables that are "basicInfo" to reflect the name change to "definedProperty"?

**Review by tenbradley — APPROVED** [review]

**Pull request merged**